### DISCUSSION TITLE
Password Reset

### DISCUSSION THREAD

**Discussion by MUzairS15** [ory/kratos#3016] [Q&A]
When I initiate an account recovery flow through code verification , after entering the code I am asked to set a new password.
The bug here is, even though I don’t set a password if I go to login, access is given. Seems like session is issued when user initiates account recovery flow (and only setting a new password remains).
Is this a bug or desired behaviour?

**Comment by vinckr** [comment]
Hello @MUzairS15,
what kind of UI are you using? This is the default behaviour in the reference UI. 
I think to get the behaviour you want (user is only able to login _after_ setting a new password) you would have to use a custom UI.

**Reply by MUzairS15** [reply]
How is this controlled via UI? I am able to login this means a session cookie has been set.
I am using html templates for UI.

**Comment by rmuratov** [comment]
As I understand this is desired behavior, though it was confusing for me also.

So after you click the recovery link you automatically log in and get "privileged session" so that you can set new password.

Here is the docs https://www.ory.sh/docs/kratos/self-service/flows/account-recovery-password-reset
